test/unit/Encoding/MicrosecondBasedDateConversionTest.php

```diff
@@ -36,8 +36,8 @@ public function dateClaimsHaveMicrosecondsOrSeconds(): void
         $formatted = $formatter->formatClaims($claims);
 
         self::assertSame(1487285080, $formatted[RegisteredClaims::ISSUED_AT]);
-        self::assertSame('1487285080.000123', $formatted[RegisteredClaims::NOT_BEFORE]);
-        self::assertSame('1487285080.123456', $formatted[RegisteredClaims::EXPIRATION_TIME]);
+        self::assertSame(1487285080.000123, $formatted[RegisteredClaims::NOT_BEFORE]);
```

Comment:
All these tests are wrong: `1487285080.000123` this literal cannot be represented precisely in memory as a floating point number, so you're comparing 2 imprecise values.

You cannot "fix" the problem in the userland, because that's how the language and IEEE754 works.